api/data.go

```diff
@@ -79,17 +82,37 @@ func (api *API) CreateOrUpdateCacheTime(ctx context.Context, w http.ResponseWrit
 		ID: id,
 	}
 
-	err := json.NewDecoder(req.Body).Decode(&docToInsertOrUpdate)
+	// Check request body not empty
+	if req.ContentLength == 0 {
+		log.Error(ctx, "createOrUpdateCacheTime endpoint: empty request body", errors.New("empty request body"))
```

Comment:
This is a tricky one: do we want to log all these 400 situations with a severity of "error", or should it be lower? Normally I would say this should probably be no higher than "info", because these are user errors. But in the case of this API, only other applications are going to be accessing it, so this situation should never happen. Maybe a "warn" is a good compromise?

Perhaps one for an ONS employee to decide.

Reply:
I would be happy with `info` here.

Reply:
Have updated